lib/apigateway.js

```diff
@@ -11,10 +11,24 @@ function sleep(s) {
 	});
 }
 
+/**
+ * Constructor method for APIGateway
+ * @param  {object}  options  - AWS sdk class options. Can contain region, profile and filename
+ */
 function APIGateway(options) {
-	// Load the credentials
 	AWS.config.region = options.region || 'us-west-1';
-	AWS.config.credentials = new AWS.SharedIniFileCredentials({profile: options.profile});
+
+	// We don't want to use the whole options object in SharedIniFileCredentials so we build up
+	// credentialsOptions
+	const credentialsOptions = {
+		profile: options.profile
+	};
+	if (options.filename) {
+		credentialsOptions.filename = options.filename;
+	}
```

Comment:
Forget about `object-assign`. But I think we should rename `profile` and `filename` to `awsProfile` and `awsFilename` so that people know it are AWS options.